lib/babe/errors.go

```diff
@@ -81,73 +79,180 @@ func (e TransactionValidityError) Error() string {
 	return fmt.Sprintf("transaction validity error: %s", e.msg)
 }
 
-func determineCustomModuleErr(res []byte) error {
-	if len(res) < 3 {
-		return errInvalidResult
-	}
-	errMsg, err := optional.NewBytes(false, nil).DecodeBytes(res[2:])
+// Dispatch Errors
+type Other string
+
+type CannotLookup struct {
+	err string
+}
+
+type BadOrigin struct {
+	err string
+}
+
+type Module struct { // add in `scale:"1"` after
+	Idx uint8
+	Err   uint8
+	Message *string
+}
+
+// Dispatch Receivers
+func (err Other) Index() uint {return 0}
+func (err CannotLookup) Index() uint {return 1}
+func (err BadOrigin) Index() uint {return 2}
+func (err Module) Index() uint {return 3}
+func (err Module) String() string {
+	return fmt.Sprintf("index: %d code: %d message: %x", err.Idx, err.Err, *err.Message)
+}
+
+// Unknown Transaction Errors
+type ValidityCannotLookup struct {
+	err string
+}
+
+type NoUnsignedValidator struct {
+	err string
+}
+
+type UnknownCustom uint8
+
+// Unknown Transaction Receivers
+func (err ValidityCannotLookup) Index() uint {return 0}
+func (err NoUnsignedValidator) Index() uint {return 1}
+func (err UnknownCustom) Index() uint {return 2}
+
+// Invalid Transaction Errors
+type Call struct {
+	err string
+}
+
+type Payment struct {
+	err string
+}
+
+type Future struct {
+	err string
+}
+
+type Stale struct {
+	err string
+}
+
+type BadProof struct {
+	err string
+}
+
+type AncientBirthBlock struct {
+	err string
+}
+
+type ExhaustsResources struct {
+	err string
+}
+
+type InvalidCustom uint8
+
+type BadMandatory struct {
+	err string
+}
+
+type MandatoryDispatch struct {
+	err string
+}
+
+// Invalid Transaction Receivers
+func (err Call) Index() uint {return 0}
+func (err Payment) Index() uint {return 1}
+func (err Future) Index() uint {return 2}
+func (err Stale) Index() uint {return 3}
+func (err BadProof) Index() uint {return 4}
+func (err AncientBirthBlock) Index() uint {return 5}
+func (err ExhaustsResources) Index() uint {return 6}
+func (err InvalidCustom) Index() uint {return 7}
+func (err BadMandatory) Index() uint {return 8}
+func (err MandatoryDispatch) Index() uint {return 9}
+
+func determineDispatchErr(res []byte) error { // This works yay!
+	var e Other
+	vdt := scale.MustNewVaryingDataType(e, CannotLookup{}, BadOrigin{}, Module{})
+	err := scale.Unmarshal(res, &vdt)
 	if err != nil {
-		return err
+		return errInvalidResult
```

Comment:
this masks the error coming from SCALE, might be nice to wrap it

Reply:
So would that be something like `return errors.New(<scale error>)`?